src/cloudimagedirectory/transform/transform.py

```diff
@@ -523,3 +523,44 @@ def run(self, data: type[Transformer]) -> list:
             # NOTE: Add /list suffix to prevent collision with "provider" folder.
             results.append(connection.DataEntry(f"v2/os/{os}/provider/list", provider_map))
         return results
+
+
+class TransformerV2ListVersionByProvider(Transformer):
+    """Generate a list for all available versions for a specific provider."""
+
+    @no_type_check
+    def run(self, data: type[Transformer]) -> list:
+        # TODO: check that its the v2 data entries.
```

Comment:
Nitpick, but I assume that this should have been "# NOTE: Verify that the data is from api v2." as well?

Reply:
Oh haha - yes I will change this :smile: